### ISSUE TITLE
the reproduced result

### ISSUE THREAD

**Issue by chenrxi** [jinkyu032/FedMLB#1]
<img width="649" alt="image" src="https://user-images.githubusercontent.com/60959000/187026114-f21fc9db-87e9-43c5-afbd-480935c31867.png">
I run the experiment in the repo and the final rusult(under Dir(0.3), 100 clients, 5% setting) is only 47.7 where is 7% lower than the original result. Do you change any Hyperparameters ?

**Comment by jinkyu032** [comment]
Hi,chenrxi.
We used 0.1 as the learning rate in our paper, but the learning rate of the github command was 1.1 due to typos.
I immediately change this to 0.1, and I'm sorry for making a mistake and confusing you.
Thank you for paying attention to our work

**Comment by chenrxi** [comment]
yes, i find this error.

**Comment by chenrxi** [comment]
How do you implement the FEDDYN code？

**Comment by jinkyu032** [comment]
Q) How do you implement the FEDDYN code？

A) When we implemented FedDyn code, since official code didn’t exist, we reproduced the code based on the algorithm on FedDyn paper.
But FedDyn official code is avaiable now, so you can utillize it from https://github.com/alpemreacar/FedDyn.
To combine FedMLB with FedDyn, we add the additional regularization term of FedMLB to the local empirical loss term(L_K(\theta)) on FedDyn.

**Comment by chenrxi** [comment]
I reproduce the SCAFFOLD on your codebase, it's validation accuracy is about 58% and quite higher than the result of FEDAVG and even the result of your FEDMLB. Have you ever tried SCAFFOLD and what do you think of the result?

**Comment by Bhongcheng** [comment]
> I reproduce the SCAFFOLD on your codebase, it's validation accuracy is about 58% and quite higher than the result of FEDAVG and even the result of your FEDMLB. Have you ever tried SCAFFOLD and what do you think of the result?

Does SCAFFOLD outperform FEDMLB in all kinds of settings? In my opinion, there is no champion method in all kinds of situation.

**Comment by chenrxi** [comment]
I only compared SCAFFOLD and FEDMLB in author's default setting(dirichlet_alpha=0.3, balanced). Your opinion is reasonable, but this setting may be one of the hard setting.

**Comment by jinkyu032** [comment]
It is awesome to hear that SCAFFOLD's local correction works well in dirichlet 0.3, although its communication cost per round is double compared to FedAvg.
And it will be great to check whether there is gain if we combine FedMLB with SCAFFOLD.
FedDyn also has higher accuracy than FedMLB when used solely.
But because it's orthogonal to FedMLB, we can combine two methods and get accuracy gain(Feddyn alone 55.78 => Combined 61.81).
SCAFFOLD is also orthogonal to FedMLB, so I expect similar effect here.

Thanks for suggesting great topic
